### ISSUE TITLE
Add test for reading/writing multiple audio formats

### ISSUE THREAD

**Issue by gavv** [roc-streaming/roc-toolkit#704]
After merging #660, add a test to [test_pump.cpp](https://github.com/roc-streaming/roc-toolkit/blob/develop/src/tests/roc_sndio/test_pump.cpp) that is the same as `write_read`, but instead of checking only wav files, it should check all supported formats (mp3, aiff, etc).

The new test should:

* invoke IBackend::discover_drivers()
* find all file formats supported by backend (use DriverInfo::type to distinguish if given driver is a file format)
* and perform the test for each of such formats

In other words, the test iterates over all backends, for each backend it iterates over all file formats, and for each such pair of backend+format it performs the same check as implemented in `write_read` test.

Doc: https://github.com/roc-streaming/roc-toolkit/blob/develop/docs/sphinx/internals/audio_backends.rst

**Comment by emmhuber** [comment]
Hello! I am new to the project and will be attempting to work on this issue.

**Comment by gavv** [comment]
@emmhuber You're welcome, #660 was merged recently.

**Comment by gavv** [comment]
@emmhuber Hi, do you still have plans on this?

**Comment by janibakin** [comment]
Hi there,

This does seem to be fairly straightforward. I can take a look, if you don't mind.
